### PR TITLE
Return of Binoculars

### PR DESCRIPTION
**Pull request by thekifake**
I believe this is actually quite a bit faster than the original method. Uses API calls now rather than just scraping information from the forum homepage.

### CODE REVIEW

File: src/injected/default_settings.json [hunk 1]
```diff
@@ -7,5 +7,6 @@
   "useHotkeys" : false,
   "stealAvatar": false,
   "answerFocus": true,
-  "restoreForum": true
+  "restoreForum": true,
+  "useBinoculars": true
```
Comment (aprzn123) [inline src/injected/default_settings.json:11]: I would suggest useBinoculars as false by default so it doesn't clutter up the page for users who don't care about it

File: src/injected/binoculars.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    b.onclick = async () => {
+      b.textContent = "Loading...";
+      let str = "Users online:";
+      const users = (await fourth.Request("forum.onlinenow")).users;
+      b.textContent = "Paste users";
+      for (let id in users) {
+        const user = users[id];
+        str += ` <b><link url="https://twocansandstring.com/users/${user.key}">${user.name}</link></b>`;
+        if (id < users.length - 1 && users.length > 2) str += ",";
+        if (id == users.length - 2) str += " and"; //...why? why does === not work and only == does? what the fuck?
```
Comment (aprzn123) [inline src/injected/binoculars.js:16]: To your comment, I'm pretty sure I saw this issue recently and it's kinda fucked up: the for-in loop makes `id` a string

**Review by aprzn123 — CHANGES_REQUESTED** [review]

File: src/injected/binoculars.js (hunk 2 above)
Comment (thekifake) [reply src/injected/binoculars.js:16]: ...What in the world? Why?

Comment (HumanoidSandvichDispenser) [reply src/injected/binoculars.js:16]: for-in loops iterate through object properties and it's not meant to be used with arrays

Comment (aprzn123) [reply src/injected/binoculars.js:16]: > ...What in the world? Why?

Javascript. Need I say more?

**Review by aprzn123 — APPROVED** [review]

**Pull request merged**